modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/IndexManager.java

```diff
@@ -281,6 +303,66 @@ boolean isIndexStale(ShpFileType indexType) {
     }
 
     /**
+     * Queries the spatial index for features available in the specified bbox, 
+     * applying the BBOX filter if specifided.
+     * 
+     * @param bbox
+     */
+    protected CloseableIterator<Data> querySpatialIndex(Envelope bbox, boolean applyEnvelopeFilter) throws DataSourceException, TreeException, IOException {
```

Comment:
No, it is a **new** method used only when a spatial part of a complex filter is visited to be solved.

https://github.com/geotools/geotools/pull/1056/files#diff-6484cabe8d1efbee3c4a0e72129755b5R102

I have not modified the current method and its behavior.